app/controllers/api/agencies_controller.rb

```diff
@@ -3,17 +3,17 @@
 module Api
   # Exposes the Agency location data
   class AgenciesController < Api::BaseController
+    before_action :permit_params
     before_action :set_agencies, only: [:index]
     before_action :set_user_location, only: [:index]
     before_action :set_agency, only: [:show]
 
     def index
-      if (@zip = search_params[:zip_code])
-        @agencies = @agencies.by_zip_code(@zip)
-      end
-      if (date = search_params[:event_date])
-        @agencies = @agencies.with_event_after(date.delete('-'))
-      end
+      by_zip(@zip)
+
+      by_event_date_after(@event_date)
```

Comment:
I think 'by_event_date_on_or_after' would be more descriptive than 'by_event_date_after'.   I realize you are following the previous model method naming but if we ever truly need event_date_after we would have to make this change.  At least something to keep in mind.

Reply:
updated